web/packages/teleterm/src/ui/uri.ts

```diff
@@ -67,6 +67,7 @@ export type ServerUri = RootClusterServerUri | LeafClusterServerUri;
 export type KubeUri = RootClusterKubeUri | LeafClusterKubeUri;
 export type DatabaseUri = RootClusterDatabaseUri | LeafClusterDatabaseUri;
 export type ClusterOrResourceUri = ResourceUri | ClusterUri;
+export type GatewayTargetUri = KubeUri | DatabaseUri;
```

Comment:
I'd move that definition to `web/packages/teleterm/src/services/tshd/types.ts`, just above the definition of the `Gateway` type. `GatewayTargetUri` is not really the same as the other URI types defined here.

I'd also keep it private without the export, at least for now.